### PR TITLE
EZP-26774: Add regression test for pure negative query handling

### PR DESCRIPTION
**Pull request by pspanja**
This PR resolves https://jira.ez.no/browse/EZP-26774

Counterpart on the Solr search engine: https://github.com/ezsystems/ezplatform-solr-search-engine/pull/81

This adds tests to expose a potentially unhandled "pure negative" query.

Quoting [Solr FAQ](https://wiki.apache.org/solr/FAQ#Why_does_.27foo_AND_-baz.27_match_docs.2C_but_.27foo_AND_.28-bar.29.27_doesn.27t_.3F):

> Boolean queries must have at least one "positive" expression (ie; MUST or SHOULD) in order to match. Solr tries to help with this, and if asked to execute a BooleanQuery that does contains only negatived clauses _at the topmost level_, it adds a match all docs query (ie: *:*)
>
> If the top level BoolenQuery contains somewhere inside of it a nested BooleanQuery which contains only negated clauses, that nested query will not be modified, and it (by definition) an't match any documents -- if it is required, that means the outer query will not match. 

For more info see:
- https://issues.apache.org/jira/browse/SOLR-80
- https://mail-archives.apache.org/mod_mbox/lucene-solr-user/201006.mbox/%3Calpine.DEB.1.10.1006011609080.29455@radix.cryptio.net%3E

### CODE REVIEW

**Comment by pspanja** [comment]
One fail exposes a bug with `MatchNone` criterion, fixed in https://github.com/ezsystems/ezplatform-solr-search-engine/pull/81/commits/c3d222eca89c41bdd979acb0c7740341edef00e2.

Ready for review, ping @andrerom @alongosz

File: bin/.travis/prepare_unittest.sh [hunk 1]
```diff
@@ -40,7 +40,7 @@ COMPOSER_UPDATE=""
 # solr package search API integration tests
 if [ "$TEST_CONFIG" = "phpunit-integration-legacy-solr.xml" ] ; then
     echo "> Require ezsystems/ezplatform-solr-search-engine:^1.1.0@dev"
-    composer require --no-update ezsystems/ezplatform-solr-search-engine:^1.1.0@dev
+    composer require --no-update ezsystems/ezplatform-solr-search-engine dev-switch-internal-query-to-edismax@dev
```
Comment (andrerom) [inline bin/.travis/prepare_unittest.sh]: tmp
